airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/avro_parser.py

```diff
@@ -50,7 +50,7 @@ def _parse_data_type(self, data_type_mapping: dict, avro_schema: dict) -> dict:
                 schema_dict[i["name"]] = [data_type_mapping[dtype] for dtype in data_type]
             # TODO: Figure out a better way to handle nested records. Currently a nested record is returned as a string
```

Comment:
Is this solved and therefore we should remove the TODO?

Reply:
@maxi297 well first of all the data type of our records should match the schema that is discovered, otherwise we will have normalization issues. As for now, nested data is emitted as is - an object inside another object. At the same time we declare it to be a string.. So we have to either modify the emitted data or the stream schema. This PR is rather a short term solution, a hotfix to unblock other PRs, so I decided to modify stream schema. As a long term solution, I believe, few more things should be done:
 - improve avro to json schema convertion (so we could be more precise in description of nested objects, arrays etc.)
 - implement json to avro schema convertion
 - use the json-converted-to-avro schema when reading the file, so that, for instance, we could pass a user enforced schema and read that nested object as a string
 
 So, answering your questions - yes, we should evaluate the `object` to be more descriptive, but I believe this should be done in a separate PR. This also means that the `todo` comment is not resolved

Reply:
Perfect! Do we have an issue for this? Should it be added to the comment?

Reply:
Sure, done!